pyproject.toml

```diff
@@ -0,0 +1,108 @@
+    {name="Joren Hammudoglu", email="jhammudoglu@gmail.com"},
+    {name="Nathaniel Starkman", email="nstarman@users.noreply.github.com"}
+  ]
+  classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python",
+    "Topic :: Scientific/Engineering",
+    "Typing :: Typed",
+  ]
+  dependencies = []
+
+  [project.urls]
+    Changelog     = "https://github.com/data-apis/array-api-typing/releases"
+    Homepage      = "https://github.com/data-apis/array-api-typing"
+
+
+[build-system]
+  requires      = ["hatch-vcs", "hatchling"]
+  build-backend = "hatchling.build"
+
+
+[dependency-groups]
+test = [
+  "pytest>=8.3.3",
+  "pytest-cov >=3",
+  "pytest-github-actions-annotate-failures",
+  "sybil>=8.0.0",
+]
+
+
+[tool.hatch]
+  build.hooks.vcs.version-file = "src/array_api_typing/_version.py"
+  version.source               = "vcs"
+
+
+[tool.pytest.ini_options]
+  addopts = [
+    "--showlocals",
+    "--strict-config",
+    "--strict-markers",
+    "-p no:doctest",    # using sybil
+    "-ra",
+  ]
+  filterwarnings = [
+    "error",
+    # Sybil
+    "ignore:Attribute s is deprecated and will be removed in Python 3\\.14:DeprecationWarning",
+    "ignore:ast\\.Str is deprecated and will be removed in Python 3\\.14:DeprecationWarning",
+  ]
+  log_cli_level = "INFO"
+  minversion = "8.3"
+  testpaths = ["docs", "src/", "tests/"]
+  norecursedirs = ["docs/_build"]
+  xfail_strict = true
+
+
+[tool.coverage]
+  report.exclude_also = ['\.\.\.', 'if typing.TYPE_CHECKING:']
+  run.source          = ["array-api-typing"]
+  run.branch          = true
+
+
+[tool.mypy]
+  allow_incomplete_defs   = false
+  allow_untyped_defs      = false
+  enable_error_code       = ["ignore-without-code", "redundant-expr", "truthy-bool"]
+  files                   = ["src", "tests"]
+  python_version          = "3.10"
+  strict                  = true
+  warn_return_any         = true
+  warn_unreachable        = true
+  warn_unused_configs     = true
+
+
+[tool.ruff]
+  [tool.ruff.lint]
+    extend-select = ["ALL"]
+    ignore = [
+      "D203",    # 1 blank line required before class docstring
+      "D213",    # Multi-line docstring summary should start at the second line
+      "ISC001",  # Conflicts with formatter
+    ]
+  
+    [tool.ruff.lint.isort]
+        combine-as-imports = true
+        extra-standard-library = ["typing_extensions"]
+        known-local-folder = ["array_api_typing"]
+
+  [tool.ruff.format]
+    docstring-code-format = true
```

Comment:
It's probably cleaner to move the installation and configuration of the tools to separate PR's (even though at first glance this config looks very reasonable to me)

Reply:
Done.